app/models/user.rb

```diff
@@ -0,0 +1,26 @@
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, :omniauth_providers => [:google_oauth2]
+
+  validates :email,
+         presence: true,
+         uniqueness: { case_sensitive: false },
+         length: { minimum: 3, maximum: 254 }
+
+  validates :firstname, presence: true
+  validates :lastname, presence: true
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.firstname = auth.info.first_name
+      user.lastname = auth.info.last_name
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0,20]
+      user.image = auth.info.image
+      # If you are using confirmable and the provider(s) you use validate emails,
+      # uncomment the line below to skip the confirmation emails.
+      # user.skip_confirmation!
```

Comment:
not yet, I've got to talk to the customer and see if they want the user to get email confirmation